yarn-project/simulator/src/public/setup_phase_manager.ts

```diff
@@ -35,10 +35,12 @@ export class SetupPhaseManager extends AbstractPhaseManager {
     previousPublicKernelProof: Proof,
   ) {
     this.log.verbose(`Processing tx ${tx.getTxHash()}`);
+    await this.publicContractsDB.addNewContracts(tx);
     const [kernelInputs, publicKernelOutput, publicKernelProof, newUnencryptedFunctionLogs, revertReason] =
       await this.processEnqueuedPublicCalls(tx, previousPublicKernelOutput, previousPublicKernelProof).catch(
         // the abstract phase manager throws if simulation gives error in a non-revertible phase
         async err => {
+          await this.publicContractsDB.removeNewContracts(tx);
```

Comment:
Hmm, not for this PR but maybe we need to create a seperate issue.

If App logic fails, will any contracts added here get removed?

Reply:
The app logic phase manager would remove all the contract from the transaction.
I just realised that this is wrong and we need to differentiate between contracts deployed in revertible vs non-revertible so that public effects are properly accounted for. I'll add a comment about this to https://github.com/AztecProtocol/aztec-packages/issues/4712